### PR TITLE
IGAPP-297: Add build config for aschaffenburg app

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: build-configs/aschaffenburg/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  WebBuildConfigType
+} from '../BuildConfigType'
+import { ASCHAFFENBURG_ASSETS } from '../AssetsType'
+import mainImprint from './mainImprint'
+
+const APPLE_APP_ID = ''
+
+const commonAschaffenburgBuildConfig: CommonBuildConfigType = {
+  appName: 'Aschaffenburg App',
+  appIcon: 'app_icon_integreat',
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:17]: I think we should TODOs in places where not the right data is not available yet.

File: build-configs/aschaffenburg/index.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  assets: ASCHAFFENBURG_ASSETS,
+  cmsUrl: 'https://cms.integreat-app.de',
+  switchCmsUrl: 'https://cms-test.integreat-app.de',
+  shareBaseUrl: 'https://aschaffenburg.app',
+  allowedHostNames: ['cms.integreat-app.de', 'cms-test.integreat-app.de'],
+  internalLinksHijackPattern: 'https?:\\/\\/(cms(-test)?\\.integreat-app\\.de|web\\.integreat-app\\.de|integreat\\.app)(?!\\/[^/]*\\/(wp-content|wp-admin|wp-json)\\/.*).*',
+  featureFlags: {
+    pois: false,
+    newsStream: true,
+    pushNotifications: true,
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:29]: Currently we don't plan on providing this functionality for Aschaffenburg.
```suggestion
    newsStream: false,
    pushNotifications: false,
```

File: build-configs/aschaffenburg/index.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  googleServices: {
+    googleAppId: '',
+    gcmDefaultSenderId: '',
+    defaultWebClientId: '',
+    gaTrackingId: null,
+    firebaseDatabaseUrl: '',
+    googleApiKey: '',
+    googleCrashReportingApiKey: '',
+    projectId: 'aschaffenburg-2021'
+  }
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:57]: This is only necessary for push notifications. As we probably won't include this in the Aschaffenburg app, you can just set this to null:
```suggestion
  googleServices: null
```

File: build-configs/aschaffenburg/index.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    projectId: 'aschaffenburg-2021',
+    storageBucket: '',
+    isAdsEnabled: false,
+    isAnalyticsEnabled: false,
+    isAppInviteEnabled: true,
+    isGCMEnabled: true,
+    isSigninEnabled: true,
+    googleAppId: '',
+    databaseUrl: ''
+  },
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:81]: Same as above:
```suggestion
  googleServices: null,
```

File: build-configs/aschaffenburg/index.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  privacyUrls: {
+    default: 'https://integreat-app.de/datenschutz/',
+    en: 'https://integreat-app.de/en/privacy/'
+  }
+}
+
+export const androidAschaffenburgBuildConfig: AndroidBuildConfigType = {
+  ...commonAschaffenburgBuildConfig,
+  splashScreen: true,
+  applicationId: 'app.aschaffenburg',
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:47]: Not sure but I think I'd rather use `de.aschaffenburgapp` similar to malte. But let's wait for the opinion of @maxammann

File: build-configs/aschaffenburg/index.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    firebaseDatabaseUrl: '',
+    googleApiKey: '',
+    googleCrashReportingApiKey: '',
+    projectId: 'aschaffenburg-2021'
+  }
+}
+
+export const iosAschaffenburgBuildConfig: iOSBuildConfigType = {
+  ...commonAschaffenburgBuildConfig,
+  bundleIdentifier: 'app.aschaffenburg',
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:62]: Same as above: Not sure but I think I'd rather use `de.aschaffenburgapp` similar to malte

File: build-configs/aschaffenburg/index.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  },
+  privacyUrls: {
+    default: 'https://integreat-app.de/datenschutz/',
+    en: 'https://integreat-app.de/en/privacy/'
+  }
+}
+
+export const androidAschaffenburgBuildConfig: AndroidBuildConfigType = {
+  ...commonAschaffenburgBuildConfig,
+  splashScreen: true,
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:46]: Similar to malte this should be false as we don't want to design a separate splash screen :D

File: build-configs/aschaffenburg/index.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    appLogo: '/app-logo.png',
+    locationMarker: '/location-marker.svg',
+    appleTouchIcon: '/apple-touch-icon.png',
+    socialMediaPreview: '/social-media-preview.png',
+    favicons: '/favicons/'
+  },
+  splashScreen: {
+    backgroundColor: lightTheme.colors.themeColor,
+    imageUrl: '/app-icon-cornered.jpg'
+  }
```
Comment (klinzo) [inline build-configs/aschaffenburg/index.js:100]: Same as above, we probably won't design a splash screen.

File: build-configs/aschaffenburg/mainImprint.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
+const mainImprint = `
```
Comment (klinzo) [inline build-configs/aschaffenburg/mainImprint.js:1]: Probably it would be good to add a TODO.

**Review by klinzo — CHANGES_REQUESTED** [review]

File: native/package.json [hunk 10]
```diff
@@ -7,13 +7,16 @@
     "start:integreat-test-cms": "cross-env BUILD_CONFIG_NAME='integreat-test-cms' react-native start --reset-cache",
     "start:integreat": "cross-env BUILD_CONFIG_NAME='integreat' react-native start --reset-cache",
     "start:malte": "cross-env BUILD_CONFIG_NAME='malte' react-native start --reset-cache",
+    "start:aschaffenburg": "cross-env BUILD_CONFIG_NAME='aschaffenburg' react-native start --reset-cache",
```
Comment (klinzo) [inline native/package.json:10]: Not sure if we rather should use a abbreviation for these commands to make them shorter. Not sure about whether there is a reasonable one though.

Comment (SteffiStuffel) [reply native/package.json]: changed it to abapp

File: build-configs/aschaffenburg/index.js (hunk 6 above)
Comment (SteffiStuffel) [reply build-configs/aschaffenburg/index.js:53]: In the Jira ticket it is specified:

App bundle id (app.aschaffenburg)
App domain (aschaffenburg.app)

**Review by klinzo — APPROVED** [review]

File: build-configs/aschaffenburg/index.js (hunk 5 above)
Comment (maxammann) [reply build-configs/aschaffenburg/index.js:47]: app.aschaffenburg is fine as aschaffenburg.app will be the offical app domain

File: build-configs/aschaffenburg/index.js (hunk 6 above)
Comment (maxammann) [reply build-configs/aschaffenburg/index.js:53]: app.aschaffenburg is fine as aschaffenburg.app will be the offical app domain

Comment (klinzo) [reply build-configs/aschaffenburg/index.js:53]: Then we did it wrong with malte :D But fine for me

File: native/package.json [hunk 11]
```diff
@@ -7,13 +7,16 @@
     "start:integreat-test-cms": "cross-env BUILD_CONFIG_NAME='integreat-test-cms' react-native start --reset-cache",
     "start:integreat": "cross-env BUILD_CONFIG_NAME='integreat' react-native start --reset-cache",
     "start:malte": "cross-env BUILD_CONFIG_NAME='malte' react-native start --reset-cache",
+    "start:abapp": "cross-env BUILD_CONFIG_NAME='aschaffenburg' react-native start --reset-cache",
```
Comment (maxammann) [inline native/package.json:10]: I think we should use start:aschaffenburg here. At least I dont have a problem with the length.

**Review by maxammann — APPROVED** [review]

File: native/package.json (hunk 11 above)
Comment (maxammann) [reply native/package.json:10]: Also abapp is short for aschaffenburg app not aschaffenburg :sweat_smile:

**Comment by klinzo** [comment]
I think you still have to add the assets here: https://github.com/Integreat/integreat-app/blob/main/native/src/modules/app/constants/buildConfig.js

**Comment by SteffiStuffel** [comment]
> @SteffiStuffel I think you still have to add the assets here: https://github.com/Integreat/integreat-app/blob/main/native/src/modules/app/constants/buildConfig.js

Yeah good point. Added this.

**Pull request merged**